plutus-use-cases/src/Language/Plutus/Coordination/Plutus.hs

```diff
@@ -1,52 +1,73 @@
+                PubKey(..)
               , Value
-              , Tx(..)
+              , Height
               , TxIn(..)
               , TxOut(..)
-              , mkAddress
               , TxOutRef(..)
-              , standardTxFee
-              , txOutValue
-              , txOutDataScript
-              , txOutValidatorScriptHash
-              -- * API operations
-              , TxM
-              , Hash
-              , hash
-              , Redeemer
-              , Validator
-              , DataScript
-              , PlutusTx(..)
-              , unitPLC
-              , BlockHeight
+              , TxId
+              -- * Pending transactions
               , PendingTx(..)
-              , submitTransaction
-              , assert
-              , lookupMyKeyPair
-              , lookupMyPubKey
-              , createPayment
-              , txInSign
-              , Range(..)
-              , EventTrigger(..)
+              , PendingTxOut(..)
+              , PendingTxIn(..)
+              -- * Oracles
               , Signed(..)
               , OracleValue(..)
               ) where
 
-import           Control.Applicative              (Alternative (..))
-import           Control.Monad.State              (State)
-import           Control.Monad.Trans.Maybe        (MaybeT (..))
-import           Language.Plutus.CoreToPLC.Plugin (PlcCode, plc)
+import           Wallet.API  (PubKey (..))
+import           Wallet.UTXO (TxId, TxIn (..), TxOut (..), TxOutRef (..))
 
-newtype Signed a = Signed (PubKey, a)
+-- | Output of a pending transaction.
+data PendingTxOut d = PendingTxOut {
+    pendingTxOutValue :: !Value,
+    pendingTxOutData  :: !d -- ^ The data script of the pending transaction output (see note [Script types in pending transactions])
+    }
+
+-- | Input of a pending transaction.
+data PendingTxIn r = PendingTxIn {
+    pendingTxInRef         :: !(TxOutRef Hash),
+    pendingTxInRefRedeemer :: !r -- ^ The redeemer of the pending transaction input (see note [Script types in pending transactions])
+    }
+
+-- | A pending transaction as seen by validator scripts.
+data PendingTx r d = PendingTx {
+    pendingTxCurrentInput :: (PendingTxIn r, Value), -- ^ The input we are validating
+    pendingTxOtherInputs  :: [(PendingTxIn r, Value)], -- ^ Other transaction inputs (they will be validated separately but we can look at their redeemer data and coin value)
+    pendingTxOutputs      :: [PendingTxOut d],
+    pendingTxForge        :: !Value,
+    pendingTxFee          :: !Value,
+    pendingTxBlockHeight  :: !Height
+    }
+
+{- Note [Script types in pending transactions]
+
+To validate a transaction, we have to evaluate the validation script of each of
+the transaction's inputs. The validation script sees the data of the
+transaction output it validates, and the redeemer of the transaction input of
+the transaction that consumes it.
+
+In addition, the validation script also needs information on the transaction as
```

Comment:
> did we decide whether this was the right way? 

I don't think an official decision has been made, these are just my own thoughts on how it could work